src/interrupts/mod.rs

```diff
@@ -100,36 +101,57 @@ lazy_static! {
+            .set_privilege_level(PrivilegeLevel::Ring3);
         idt.device_not_available
-            .set_handler_fn(device_not_available_handler);
-        idt.invalid_tss.set_handler_fn(invalid_tss_handler);
+            .set_handler_fn(device_not_available_handler)
+            .set_privilege_level(PrivilegeLevel::Ring3);
+        idt.invalid_tss.set_handler_fn(invalid_tss_handler)
+            .set_privilege_level(PrivilegeLevel::Ring3);
         idt.segment_not_present
-            .set_handler_fn(segment_not_present_handler);
+            .set_handler_fn(segment_not_present_handler)
+            .set_privilege_level(PrivilegeLevel::Ring3);
         idt.stack_segment_fault
-            .set_handler_fn(stack_segment_fault_handler);
+            .set_handler_fn(stack_segment_fault_handler)
+            .set_privilege_level(PrivilegeLevel::Ring3);
         idt.general_protection_fault
-            .set_handler_fn(general_protection_fault_handler);
-        idt.page_fault.set_handler_fn(page_fault_handler);
+            .set_handler_fn(general_protection_fault_handler)
+            .set_privilege_level(PrivilegeLevel::Ring3);
+        idt.page_fault.set_handler_fn(page_fault_handler)
+            .set_privilege_level(PrivilegeLevel::Ring3);
         idt.x87_floating_point
-            .set_handler_fn(x87_floating_point_handler);
-        idt.alignment_check.set_handler_fn(alignment_check_handler);
+            .set_handler_fn(x87_floating_point_handler)
+            .set_privilege_level(PrivilegeLevel::Ring3);
+        idt.alignment_check.set_handler_fn(alignment_check_handler)
+            .set_privilege_level(PrivilegeLevel::Ring3);
         idt.simd_floating_point
-            .set_handler_fn(simd_floating_point_handler);
-        idt.virtualization.set_handler_fn(virtualization_handler);
+            .set_handler_fn(simd_floating_point_handler)
+            .set_privilege_level(PrivilegeLevel::Ring3);
+        idt.virtualization.set_handler_fn(virtualization_handler)
+            .set_privilege_level(PrivilegeLevel::Ring3);
         idt.security_exception
-            .set_handler_fn(security_exception_handler);
-        idt.timer.set_handler_fn(saveRegisters!(timer_interrupt_handler));
-        idt[InterruptIndex::Keyboard.as_usize()].set_handler_fn(keyboard_interrupt_handler);
-        idt.syscall.set_handler_fn(syscalls::naked_syscall_dispatch);
+            .set_handler_fn(security_exception_handler)
+            .set_privilege_level(PrivilegeLevel::Ring3);
+        idt.timer.set_handler_fn(saveRegisters!(timer_interrupt_handler))
+            .set_privilege_level(PrivilegeLevel::Ring3);
+        idt[InterruptIndex::Keyboard.as_usize()].set_handler_fn(keyboard_interrupt_handler)
+            .set_privilege_level(PrivilegeLevel::Ring3);
+        idt.syscall.set_handler_fn(syscalls::naked_syscall_dispatch)
+                    .set_privilege_level(PrivilegeLevel::Ring3);
```

Comment:
Maybe we could wrap all this code into a proper function. But for development sakes, it is ofc ok!